solr/contrib/ltr/src/test/org/apache/solr/ltr/TestLTROnSolrCloud.java

```diff
@@ -232,32 +230,47 @@ private void createCollection(String name, String config, int numShards, int num
     solrCluster.waitForActiveCollection(name, numShards, numShards * numReplicas);
   }
 
-
   void indexDocument(String collection, String id, String title, String description, int popularity)
     throws Exception{
     SolrInputDocument doc = new SolrInputDocument();
     doc.setField("id", id);
     doc.setField("title", title);
     doc.setField("description", description);
     doc.setField("popularity", popularity);
+    if(popularity != 1) {
+      // check that empty values will be read as default
+      doc.setField("dvIntField", popularity);
+      doc.setField("dvLongField", popularity);
+      doc.setField("dvFloatField", ((float) popularity) / 10);
+      doc.setField("dvDoubleField", ((double) popularity) / 10);
+      doc.setField("dvStrNumField", popularity);
+      doc.setField("dvStrBoolField", popularity % 2 == 0 ? "T" : "F");
+    }
     solrCluster.getSolrClient().add(collection, doc);
   }
 
   private void indexDocuments(final String collection)
        throws Exception {
     final int collectionSize = 8;
-    for (int docId = 1; docId <= collectionSize;  docId++) {
+    // put documents in reversed order to check that advanceExact is working correctly
+    for (int docId = collectionSize; docId >= 1;  docId--) {
       final int popularity = docId;
       indexDocument(collection, String.valueOf(docId), "a1", "bloom", popularity);
+      if(docId == collectionSize / 2) {
+        // commit in the middle in order to check that everything works fine for multi-segment case
+        solrCluster.getSolrClient().commit(collection);
+      }
```

Comment:
It's great to expand the test coverage here to check that the multi-segment case works. However _always_ having the commit in the middle also changes what the existing test does i.e. the single-segment or fewer segments scenario. A _"best of both worlds"_ way could be use of randomisation i.e. only _sometimes_ commit in the middle e.g.

```
  if (random.nextBoolean()) solrCluster.getSolrClient().commit(collection);
```

which is supported by the test framework.

Reply:
Thanks for the suggestion with nextBoolean(), I didn't know that. Used it in 4559415.